frontend/site-registry/src/app/features/details/SiteDetails.tsx

```diff
@@ -261,30 +261,56 @@ const SiteDetails = () => {
     if (id) {
       dispatch(resetSaveSiteDetails(null));
       dispatch(setupSiteIdForSaving(id));
-      Promise.all([
-        dispatch(fetchSnapshots(id ?? '')),
-        dispatch(getBannerType(id ?? '')),
-        dispatch(fetchMinistryContact('EMP')),
-        dispatch(fetchNotationClassCd()),
-        dispatch(fetchNotationTypeCd()),
-        dispatch(fetchNotationParticipantRoleCd()),
-        dispatch(fetchParticipantRoleCd()),
-        dispatch(
-          fetchSiteParticipants({ siteId: id ?? '', showPending: false }),
-        ),
-        dispatch(
-          fetchNotationParticipants({ siteId: id ?? '', showPending: false }),
-        ),
-        // should be based on condition for External and Internal User.
-        dispatch(fetchSitesDetails({ siteId: id ?? '', showPending: false })),
-        // dispatch(fetchNotationParticipants({ siteId: id ?? '', showPending: false})),
-      ])
-        .then(() => {
-          setIsLoading(false); // Set loading state to false after all API calls are resolved
-        })
-        .catch((error) => {
-          console.error('Error fetching data:', error);
-        });
+      if (userType === UserType.External) {
```

Comment:
You can only add `UserType.External ? dispatch(getBannerType(id ?? '')) : Promise.resolve(),` to make sure the the method is dispatched only for external users instead of replicating the whole section.